src/StaticCaching/DefaultInvalidator.php

```diff
@@ -22,58 +26,90 @@ public function __construct(Cacher $cacher, $rules = [])
         $this->rules = $rules;
     }
 
-    public function invalidate($item)
+    public function invalidate($item): void
     {
         if ($this->rules === 'all') {
-            return $this->cacher->flush();
+            $this->cacher->flush();
+
+            return;
         }
 
         if ($item instanceof Entry) {
             $this->invalidateEntryUrls($item);
-        } elseif ($item instanceof Term) {
+        } elseif ($item instanceof LocalizedTerm) {
             $this->invalidateTermUrls($item);
         } elseif ($item instanceof Nav) {
             $this->invalidateNavUrls($item);
-        } elseif ($item instanceof GlobalSet) {
+        } elseif ($item instanceof NavTree) {
+            $this->invalidateNavTreeUrls($item);
+        } elseif ($item instanceof Variables) {
             $this->invalidateGlobalUrls($item);
         } elseif ($item instanceof Collection) {
             $this->invalidateCollectionUrls($item);
+        } elseif ($item instanceof CollectionTree) {
+            $this->invalidateCollectionTreeUrls($item);
         } elseif ($item instanceof Asset) {
             $this->invalidateAssetUrls($item);
         } elseif ($item instanceof Form) {
             $this->invalidateFormUrls($item);
         }
     }
 
-    protected function invalidateFormUrls($form)
+    protected function invalidateFormUrls($form): void
     {
-        $this->cacher->invalidateUrls(
-            Arr::get($this->rules, "forms.{$form->handle()}.urls")
-        );
+        $rules = collect(Arr::get($this->rules, "forms.{$form->handle()}.urls"));
+
+        $rules
+            ->filter(fn (string $rule) => $this->isAbsoluteUrl($rule))
+            ->each(fn (string $rule) => $this->cacher->invalidateUrl($rule));
+
+        Site::all()->each(function ($site) use ($rules) {
+            $rules
+                ->reject(fn (string $rule) => $this->isAbsoluteUrl($rule))
+                ->each(fn (string $rule) => $this->cacher->invalidateUrl(Str::removeRight($site->url(), '/').Str::ensureLeft($rule, '/')));
+        });
```

Comment:
Would you be able to combine these all into a single array that gets passed into cacher->invalidateUrls() one time?

```php
$this->cacher->invalidateUrls([
    ...$hardCodedAbsoluteUrls,
    ...$relativeUrlsConvertedToAbsoluteUrlsInEachSite
]);
```